### PR TITLE
Update draco3d types

### PR DESCRIPTION
**Pull request by servinlp**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
https://github.com/google/draco/blob/master/src/draco/mesh/mesh.h#L52
https://github.com/google/draco/blob/master/src/draco/point_cloud/point_cloud.h#L29

### CODE REVIEW

File: types/draco3d/index.d.ts [hunk 1]
```diff
@@ -105,8 +114,8 @@ export interface Decoder {
     GetAttributeUInt16ForAllPoints: (mesh: Mesh, attribute: Attribute, array: DracoArray) => void;
     GetAttributeUInt32ForAllPoints: (mesh: Mesh, attribute: Attribute, array: DracoArray) => void;
     GetEncodedGeometryType: (buffer: DecoderBuffer) => GeometryType;
-    GetAttributeId: (mesh: Mesh, attributeType: number) => number;
-    GetAttribute: (mesh: Mesh, id: number) => Attribute;
+    GetAttributeId: (mesh: PointCloud, attributeType: number) => number;
+    GetAttribute: (mesh: PointCloud, id: number) => Attribute;
```
Comment (donmccurdy) [inline types/draco3d/index.d.ts:118]: See other comment — probably want `Mesh | PointCloud` here.

File: types/draco3d/index.d.ts [hunk 2]
```diff
@@ -147,56 +156,24 @@ export enum GeometryType {}
 // tslint:disable-next-line:no-empty-interface
 export enum DataType {}
 
-export interface Mesh {
+export interface PointCloud {
+    num_attributes: () => number;
+    num_points: () => number;
+}
+
+export interface Mesh extends PointCloud {
```
Comment (donmccurdy) [inline types/draco3d/index.d.ts:164]: Does this actually extend a PointCloud class in the Draco C++ library? If not I'd prefer not to represent it as a subclass here... mind separating these? I think it's OK to duplicate the two methods above.

**Review by donmccurdy — COMMENTED** [review]
Thanks for these updates! One suggestion on the Mesh / PointCloud classes, otherwise looks good to me.

File: types/draco3d/index.d.ts (hunk 2 above)
Comment (servinlp) [reply types/draco3d/index.d.ts:164]: Yes, from the source `Mesh is just an extended PointCloud with extra connectivity data that defines`
https://github.com/google/draco/blob/master/src/draco/mesh/mesh.h#L52

File: types/draco3d/index.d.ts (hunk 1 above)
Comment (servinlp) [reply types/draco3d/index.d.ts:118]: This is based on the following
https://github.com/google/draco/blob/master/src/draco/javascript/emscripten/decoder_webidl_wrapper.h#L130
https://github.com/google/draco/blob/master/src/draco/javascript/emscripten/decoder_webidl_wrapper.cc#L125

File: types/draco3d/index.d.ts (hunk 2 above)
Comment (donmccurdy) [reply types/draco3d/index.d.ts:164]: Huh that's interesting! Makes sense to do the same here then thanks. 👍🏻

**Review by donmccurdy — APPROVED** [review]

**Comment by servinlp** [comment]
Ready to merge

**Pull request merged**